src/lib/engine/index.ts

```diff
@@ -112,19 +132,11 @@ export function runGame(code: string, canvas: HTMLCanvasElement, onPageError: (e
 	}
 
     const engineAPIKeys = Object.keys(api);
-    try {
-        const transformResult = Babel.transform(code, {
-            plugins: [TransformDetectInfiniteLoop, BuildDuplicateFunctionDetector(engineAPIKeys)],
-            retainLines: true
-        });
-        logInfo.value = [];
-        const fn = new Function(...engineAPIKeys, transformResult.code!);
-        fn(...Object.values(api));
-        return { error: null, cleanup };
-    } catch (error: any) {
-        onPageError(normalizeGameError({ kind: "runtime", error }));
-        return { error: normalizeGameError({ kind: "runtime", error }), cleanup };
-    }
+	return { error: transformAndThrowErrors(code, engineAPIKeys, (transformedCode) => {
```

Comment:
how confident are we in this change?  how carefully have you tested it, correctness seems very important here...